src/two_finger_grasp_filter.cpp

```diff
@@ -0,0 +1,160 @@
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the Univ of CO, Boulder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Dave Coleman <dave@picknik.ai>
+   Desc:   Filters grasps based on kinematic feasibility and collision
+*/
+
+// Parent Class
+#include <moveit_grasps/state_validity_callback.h>
+
+// moveit_grasps
+#include <moveit_grasps/two_finger_grasp_filter.h>
+
+// moveit
+#include <moveit/transforms/transforms.h>
+#include <moveit/collision_detection/collision_tools.h>
+
+namespace moveit_grasps
+{
+// Constructor
+TwoFingerGraspFilter::TwoFingerGraspFilter(robot_state::RobotStatePtr robot_state,
+                                           moveit_visual_tools::MoveItVisualToolsPtr& visual_tools)
+  : GraspFilter(robot_state, visual_tools)
+{
+}
+
+bool TwoFingerGraspFilter::processCandidateGrasp(IkThreadStructPtr& ik_thread_struct)
+{
+  ROS_DEBUG_STREAM_NAMED("grasp_filter.superdebug", "Checking grasp #" << ik_thread_struct->grasp_id);
+
+  // Helper pointer
+  GraspCandidatePtr& grasp_candidate = ik_thread_struct->grasp_candidates_[ik_thread_struct->grasp_id];
+
+  // Get pose
+  ik_thread_struct->ik_pose_ = grasp_candidate->grasp_.grasp_pose;
+
+  // Filter by cutting planes
+  for (std::size_t i = 0; i < cutting_planes_.size(); i++)
+  {
+    if (filterGraspByPlane(grasp_candidate, cutting_planes_[i]->pose_, cutting_planes_[i]->plane_,
+                           cutting_planes_[i]->direction_) == true)
+    {
+      grasp_candidate->grasp_filtered_by_cutting_plane_ = true;
+      return false;
+    }
+  }
+
+  // Filter by desired orientation
+  for (std::size_t i = 0; i < desired_grasp_orientations_.size(); i++)
```

Comment:
++i

Reply:
Consider range for loop. See comment above.